_plugins/sort_edtf.rb

```diff
@@ -1,10 +1,46 @@
 module Jekyll
-    module SortEDTF
-        def sort_edtf(array_of_strings)
-            sorted = array_of_strings.map { |str| str.gsub('X', '0') }
-            sorted.sort_by { |str| str[/\d+/].to_i }
-        end
+  module SortEDTF
+    def clean_date_string(str)
+      str.gsub(/[Xu]/, '0')
     end
+
+    def compute_numeric_value(cleaned_str)
```

Comment:
@koilebeit Minor remark: the function does not say clearly what it does (and the input is also not clearly labeled)